clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/CachingKafkaConsumerFactory.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.kafka.client;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+
+/**
+ * TODO.
+ *
+ * @param <K> The type for the record key serialization.
+ * @param <V> The type for the record value serialization.
+ */
+public class CachingKafkaConsumerFactory<K, V> implements KafkaConsumerFactory<K, V> {
```

Comment:
I added a quite generic Kafka consumer for AT LEAST ONCE consumption: `AbstractAtLeastOneKafkaConsumer` from the PR #2420 (it does not require the factory). Would it FYPOV make sense to use it here?